scalyr_agent/agent_main.py

```diff
@@ -794,8 +909,15 @@ def __detailed_status(
         # We wait for five seconds at most to get the status.
         deadline = time.time() + 5
 
+        last_debug_stat_time = 0
         # Now loop until we see it show up.
         while True:
+            # Capture debug stats every second while waiting.
+            if debug and (last_debug_stat_time + 1 < time.time()):
```

Comment:
I wonder if we should make that value configurable via command line options because in case there is indeed a race condition, we may need to use lower value (e.g. 0.5 seconds or less) to try to find offending stray process which may be causing a race condition.

Reply:
EDIT: Never mind, I noticed below it only works with health check health option right now so this change may not be needed.